src/spec/create_extension_spec.py

```diff
@@ -0,0 +1,531 @@
+                    "(v2x, v2y, v2z). "
+                    "See 'contact_plane_axes' in "
+                    "https://probeinterface.readthedocs.io/en/main/format_spec.html for more details."
+                ),
+                dtype="float",
+                dims=[["num_contacts", "v1, v2", "x, y"], ["num_contacts", "v1, v2", "x, y, z"]],
+                shape=[[None, 2, 2], [None, 2, 3]],
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="shape",
+                neurodata_type_inc="VectorData",
+                doc="Shape of the contact; e.g. 'circle'",
+                dtype="text",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="radius_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Radius of a circular contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="width_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Width of a rectangular or square contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="height_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Height of a rectangular contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+        ],
+    )
+
+    probe = NWBGroupSpec(
+        neurodata_type_def="Probe",
+        neurodata_type_inc="Device",
+        doc="Specific instance of a neural probe object.",
+        links=[
+            NWBLinkSpec(
+                name="probe_model",
+                doc="The model of the probe used to record the data.",
+                target_type="ProbeModel",
+            ),
+        ],
+        attributes=[
+            NWBAttributeSpec(
+                name="identifier",
+                doc="Identifier of the probe, usually the serial number.",
+                dtype="text",
+                required=False,
+            ),
+        ],
+    )
+
+    probe_model = NWBGroupSpec(
+        neurodata_type_def="ProbeModel",
+        neurodata_type_inc="Device",
+        doc=(
+            "Neural probe object, compatible with the ProbeInterface specification. The name of the object should "
+            'be the model name of the probe, e.g., "Neuropixels 1.0".'
+        ),
+        groups=[
+            NWBGroupSpec(
+                name="contacts_table",
+                neurodata_type_inc="ContactsTable",
+                doc="Neural probe contacts, compatible with the ProbeInterface specification",
+            ),
+        ],
+        attributes=[
+            # inherits name, description, manufacturer from Device
+            NWBAttributeSpec(name="ndim", doc="dimension of the probe", dtype="int", default_value=2),
+            NWBAttributeSpec(
```

Comment:
Could we modify ProbeModel so name is just the model by default unless the user specify it?

Reply:
Good idea. I'll make that change.